File: 프로그래머스/lv2/43165. 타겟 넘버/타겟 넘버.py
def solution(numbers, target):
    cnt = 0
    list_n =  [0]
    for n in numbers:
        tmp = []
        for l in list_n:
            tmp.append(l + n)
            tmp.append(l - n)
        list_n = tmp
    return list_n.count(target)
        

File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.java
import java.util.*;

class Solution {
    public int solution(int[] numbers, int target) {
        ArrayList<Integer> arr = new ArrayList<>();
        arr.add(0);
        
        for(int n : numbers){
            ArrayList<Integer> tmp = new ArrayList<>(); 
            for(int a : arr){
                tmp.add(a + n);
                tmp.add(a - n);
            }
            arr.clear();
            arr.addAll(tmp);
        }
        
        int answer = 0;
        for(int a : arr){
            if(a == target) answer++;
        }
        return answer;
    }
}